src/model/model.nit

```diff
@@ -1201,6 +1273,265 @@ abstract class MType
 	end
 end
 
+# A type that represents an operation over a set of types.
+abstract class MTypeSet[E: MType]
+	super MType
+
+	# The context in which this type was created.
+	#
+	# Used to populate `model` and look for `Object`.
+	#
+	# TODO: Remove this attribute once `as_notnull` don’t need it.
+	var mmodule: MModule
```

Comment:
My idea is to move it as a parameter in a future PR (already done, just not submitted). The only alternative I see is to change the way `Object` is introduced in the model.